app/src/main/java/de/xikolo/models/Video.kt

```diff
@@ -0,0 +1,131 @@
+
+    var lecturerStream: VideoStream? = null
+
+    var slidesStream: VideoStream? = null
+
+    var slidesUrl: String? = null
+
+    var slidesSize: Int = 0
+
+    var audioUrl: String? = null
+
+    var audioSize: Int = 0
+
+    var transcriptUrl: String? = null
+
+    var transcriptSize: Int = 0
+
+    var thumbnailUrl: String? = null
+
+    var subtitles = RealmList<VideoSubtitles>()
+
+    @Ignore
+    var item: Item? = findForContent(id)
+
+    // local field
+    var progress = 0
+
+    val streamToPlay: VideoStream
+        get() {
+            return if (singleStream != null && (singleStream!!.hdUrl != null || singleStream!!.sdUrl != null)) {
+                singleStream as VideoStream
+            } else if (lecturerStream != null && (lecturerStream!!.hdUrl != null || lecturerStream!!.sdUrl != null)) {
+                lecturerStream as VideoStream
+            } else slidesStream as VideoStream
+        }
+
+    @JsonApi(type = "videos")
+    class JsonModel : Resource(), RealmAdapter<Video> {
+
+        var summary: String? = null
+        var duration: Int = 0
+
+        @field:Json(name = "single_stream")
+        var singleStream: VideoStream? = null
+
+        @field:Json(name = "lecturer_stream")
+        var lecturerStream: VideoStream? = null
+
+        @field:Json(name = "slides_stream")
+        var slidesStream: VideoStream? = null
+
+        @field:Json(name = "slides_url")
+        var slidesUrl: String? = null
+
+        @field:Json(name = "slides_size")
+        var slidesSize: Int = 0
+
+        @field:Json(name = "audio_url")
+        var audioUrl: String? = null
+
+        @field:Json(name = "audio_size")
+        var audioSize: Int = 0
+
+        @field:Json(name = "transcript_url")
+        var transcriptUrl: String? = null
+
+        @field:Json(name = "transcript_size")
+        var transcriptSize: Int = 0
+
+        @field:Json(name = "thumbnail_url")
+        var thumbnailUrl: String? = null
+
+        var subtitles: List<VideoSubtitles>? = null
+
+        override fun convertToRealmObject(): Video {
+            val video = Video()
+            video.id = id
+            video.summary = summary
+            video.duration = duration
+            video.singleStream = singleStream
+            video.lecturerStream = lecturerStream
+            video.slidesStream = slidesStream
+            video.slidesUrl = slidesUrl
+            video.slidesSize = slidesSize
+            video.audioUrl = audioUrl
+            video.audioSize = audioSize
+            video.transcriptUrl = transcriptUrl
+            video.transcriptSize = transcriptSize
+            video.thumbnailUrl = thumbnailUrl
+            video.subtitles.addAll(subtitles!!)
+            return video
+        }
+    }
+
+    fun getSubtitlesAsString(): String {
+        return subtitles.map {
+            LanguageUtil.toLocaleName(it.language)
+        }.joinToString(", ", "")
+    }
+
+    val isSummaryAvailable: Boolean
```

Comment:
So I guess `val`s with a getter are not being considered by Realm? Please move `item` and `streamToPlay` also down to these "helper" fields

Reply:
The `@Ignore` tag is then not needed